app/Http/Controllers/Api/ChangeLocaleController.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+
+class ChangeLocaleController extends Controller
+{
+    public function __invoke(Request $request, string $locale): JsonResponse
+    {
+        if (in_array($locale, config("app.supported_locales"), true)) {
+            app()->setLocale($locale);
+
+            return response()->json([
+                "message" => __("Language has been changed."),
+            ]);
+        }
+
+        return response()->json([
+            "message" => __("Error changing the language."),
+        ]);
+    }
+}
```

Comment:
Do we need changing locale in API?
REST API should be stateless and what I see, we have not storing that information in any user profile.